src/modules/Weth.ts

```diff
@@ -0,0 +1,60 @@
+import BigNumber from 'bignumber.js';
+import { Contract } from 'web3-eth-contract';
+
+import {
+  BigNumberable,
+  SendOptions,
+  TxResult,
+  address,
+} from '../lib/types';
+import { Contracts } from './Contracts';
+import { Token } from './Token';
+
+/**
+ * Client module for interacting with the WETH9 token contract.
+ */
+export class Weth extends Token {
+  private weth: Contract;
+
+  constructor(
+    contracts: Contracts,
+  ) {
+    super(contracts);
+    this.weth = contracts.weth;
+  }
+
+  public get address(): string {
+    return this.weth.options.address;
+  }
+
+  public async wrap(
+    ownerAddress: address,
```

Comment:
nit: I believe for consistency we shouldn't have `ownerAddress` be explicitly given for `wrap()` and `unwrap()`. We should rely on the user to supply it in the `options`

Reply:
I was considering whether it should be consistent with Solo’s weth module but otherwise I agree. What do you think about amount as opposed to passing in as value in options?

Reply:
I think that is a good question. I think amount should still be an explicit parameter, although it's difficult for me to explain why it should be different than `from`.

Perhaps the "safest" solution is to do error-checking in this function that if `options.value` is supplied, that it doesn't contradict the amount already given? Not sure